pkg/api/options.go

```diff
@@ -34,3 +38,32 @@ func FullObjects(fullObjects bool) ListOption {
 func AutoTag(tags ...string) CreateOption {
 	return internal.AutoTagOption(tags)
 }
+
+// EnvironmentOption can be used to configure an alternative environment path
+// segment for a given API group
+type EnvironmentOption struct {
+	APIGroup       string
+	EnvPathSegment string
+	Override       bool
+}
+
+// ApplyToAny applies the EnvironmentOption to any request type
+func (o EnvironmentOption) ApplyToAny(opts types.Options) error {
+	return opts.Set(fmt.Sprintf("environment/%s", o.APIGroup), o.EnvPathSegment, o.Override)
+}
+
+// GetEnvironmentPathSegment retrieves the environment path segment of a given API group
+// or the provided defaultValue if no environment override is set
+func GetEnvironmentPathSegment(ctx context.Context, apiGroup, defaultValue string) (string, error) {
+	if options, err := types.OptionsFromContext(ctx); err != nil {
+		return "", err
+	} else if env, err := options.Get(fmt.Sprintf("environment/%s", apiGroup)); errors.Is(err, types.ErrKeyNotSet) {
+		return defaultValue, nil
+	} else if err != nil {
+		return "", err
+	} else if envString, ok := env.(string); !ok {
+		return "", ErrEnvironmentSegmentNotTypeString
```

Comment:
if the comment above is accepted, this is the last branch that could return an error - could it be sensible to have a `environments` variable on `types.commonOptions` that can only hold strings, removing the need for this function to be able to return an error, which then also helps with the now 5 returns in `apis/kubernetes/v1.endpointURL`?

Reply:
something like that, yea